reana_db/models.py

```diff
@@ -435,6 +456,41 @@ def can_transition_to(self, next_status):
         return current_transition in ALLOWED_WORKFLOW_STATUS_TRANSITIONS
 
 
+@event.listens_for(Workflow.status, "set")
+def workflow_status_change_listener(workflow, new_status, old_status, initiator):
+    """Workflow status change listener."""
+    if new_status in [
+        WorkflowStatus.finished,
+        WorkflowStatus.failed,
+    ]:
+        workflow.run_finished_at = datetime.now()
+    elif new_status in [WorkflowStatus.stopped]:
+        workflow.run_stopped_at = datetime.now()
+
+    finished_at = workflow.run_finished_at or workflow.run_stopped_at
+    if workflow.run_started_at and finished_at:
+        RESOURCE = "cpu"
+        cpu_time = finished_at - workflow.run_started_at
+        cpu_milliseconds = int(cpu_time.total_seconds() * 1000)
+        cpu_resource = Resource.query.filter_by(name=RESOURCE).one_or_none()
+        if cpu_resource:
+            from .database import Session
+
+            workflow_resource = WorkflowResource(
+                workflow_id=workflow.id_,
+                resource_id=cpu_resource.id_,
+                quantity_used=cpu_milliseconds,
+            )
+            user_resource_quota = UserResource.query.filter_by(
```

Comment:
We are assuming for now that the [admins have successfully populated the quota related tables](https://github.com/reanahub/reana/pull/416). For now, if this is not the case, we return `null` as CPU usage. What should we do?
1. Continue
    - Log what has happened, that quota was requested but null was returned as there is missing config
2. Error out
    - Maybe not if the admins are not interested in accounting usage?

Reply:
Perhaps accounting should be optional and only calculate quotas if `Resource`s and `UserResource`s are set. Otherwise, we would be forcing admins to set quota limits per user (unless we let them set some defaults).

Reply:
Why don't we have some default quota limit value (say 0) meaning "no quota"? 

If an admin wants to set quotas to users, this would be always some positive number, say 1 sec or 1 KiB. Because to ban people, admins could just revoke the token... No need to "nullify" quota for this.

The accounting, if cheap, could be done always, so that admins know what's happening... But the end user would see "no quota limit" if the limit value is the default one (0).